tests/python/test_with_dask.py

```diff
@@ -149,68 +149,46 @@ def test_dask_predict_shape_infer() -> None:
 
 
 @pytest.mark.parametrize("tree_method", ["hist", "approx"])
-def test_boost_from_prediction(tree_method: str) -> None:
-    if tree_method == 'approx':
-        pytest.xfail(reason='test_boost_from_prediction[approx] is flaky')
```

Comment:
Is the test no longer flaky?

Reply:
@hcho3 I rewrote the test.

Reply:
That doesn't mean it's no longer flaky, we will have to see in the future.